### ISSUE TITLE
[VarDumper] Notice: Undefined offset when dumping casted array from object with PHP7 

### ISSUE THREAD

**Issue by tgabi333** [symfony/symfony#19356]
I've got an Undefined offset notice here: https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/VarDumper/Cloner/VarCloner.php#L74

Simplest example: 

```php
        $obj = json_decode('{"1":1,"2":1}');
        $array = (array) $obj;
        dump($array);
```

Environment:

```
$ composer.phar show | grep symfony/symfony
symfony/symfony                      v2.8.8             The Symfony PHP framework
```

```
$ php -v
PHP 7.0.8-3+deb.sury.org~xenial+1 (cli) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.8-3+deb.sury.org~xenial+1, Copyright (c) 1999-2016, by Zend Technologies
```

**Comment by tgabi333** [comment]
Its interesting: https://3v4l.org/ACrH6

**Comment by valisj** [comment]
@tgabi333 I could throw in some thoughts here:

1) From PHP's standpoint this is not a bug. If an object is casted to an array, integer properties will become inaccessible. (http://php.net/manual/en/language.types.array.php#language.types.array.casting)

2) Passing `true` as the 2nd argument to the function call, like so: `$obj = json_decode('{"1":1,"2":1}', true);`, will give you an array where all keys are accessible.

**Comment by tgabi333** [comment]
It's okay that is a documented php behavior, But `VarDumper` should handle it and not raise an Undefined offset noties

**Comment by nicolas-grekas** [comment]
Which version are you using? I remember fixing this already but maybe we have a regression?

**Comment by tgabi333** [comment]
which is bundled in symfony 2.8.8

**Comment by nicolas-grekas** [comment]
Should be fixed by #19368

**Comment by tgabi333** [comment]
Thanks @nickolas-grekas
